api/v1alpha1/dnspolicy_types.go

```diff
@@ -56,7 +51,7 @@ const (
 )
 
 // DNSPolicySpec defines the desired state of DNSPolicy
-// +kubebuilder:validation:XValidation:rule="!(self.routingStrategy == 'loadbalanced' && !has(self.loadBalancing))",message="spec.loadBalancing is a required field when spec.routingStrategy == 'loadbalanced'"
+// +kubebuilder:validation:XValidation:rule="(!has(oldSelf.loadBalancing) || has(self.loadBalancing)) && (has(oldSelf.loadBalancing) || !has(self.loadBalancing))", message="loadBalancing is immutable"
```

Comment:
Find these validations a bit confusing to read tbh, but this just means you can't change the filed  `loadBalancing` from set to unset and vice versa after creation. Doing it on the field itself would mean you couldn't change the value of  `loadBalancing` and therefore any fields in that struct.

Tests to check all this were added.